internal/provider/provider.go

```diff
@@ -1,192 +1,220 @@
-	resp.TypeName = "bloxone"
-	resp.Version = p.version
+    resp.TypeName = "bloxone"
+    resp.Version = p.version
 }
 
 func (p *BloxOneProvider) Schema(_ context.Context, _ provider.SchemaRequest, resp *provider.SchemaResponse) {
-	resp.Schema = schema.Schema{
-		Description: "The BloxOne provider is used to interact with the resources supported by Infoblox BloxOne API.",
-		Attributes: map[string]schema.Attribute{
-			"csp_url": schema.StringAttribute{
-				MarkdownDescription: "URL for BloxOne Cloud Services Portal. Can also be configured using the `BLOXONE_CSP_URL` environment variable.",
-				Optional:            true,
-			},
-			"api_key": schema.StringAttribute{
-				MarkdownDescription: "API key for accessing the BloxOne API. Can also be configured by using the `BLOXONE_API_KEY` environment variable. https://docs.infoblox.com/space/BloxOneCloud/35430405/Configuring+User+API+Keys",
-				Optional:            true,
-			},
-		},
-	}
+    resp.Schema = schema.Schema{
+        Description: "The BloxOne provider is used to interact with the resources supported by Infoblox BloxOne API.",
+        Attributes: map[string]schema.Attribute{
+            "csp_url": schema.StringAttribute{
+                MarkdownDescription: "URL for BloxOne Cloud Services Portal. Can also be configured using the `BLOXONE_CSP_URL` environment variable.",
+                Optional:            true,
+            },
+            "api_key": schema.StringAttribute{
+                MarkdownDescription: "API key for accessing the BloxOne API. Can also be configured by using the `BLOXONE_API_KEY` environment variable. https://docs.infoblox.com/space/BloxOneCloud/35430405/Configuring+User+API+Keys",
+                Optional:            true,
+            },
+            "default_tags": schema.MapAttribute{
+                ElementType:         types.StringType,
+                MarkdownDescription: "Tags to default for all resources.",
+                Optional:            true,
+            },
+        },
+    }
 }
 
 func (p *BloxOneProvider) Configure(ctx context.Context, req provider.ConfigureRequest, resp *provider.ConfigureResponse) {
-	var data BloxOneProviderModel
+    var data BloxOneProviderModel
 
-	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+    resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
 
-	if resp.Diagnostics.HasError() {
-		return
-	}
+    var dfTags map[string]string
 
-	client, err := bloxoneclient.NewAPIClient(bloxoneclient.Configuration{
-		ClientName: fmt.Sprintf("terraform/%s#%s", p.version, p.commit),
-		APIKey:     data.APIKey.ValueString(),
-		CSPURL:     data.CSPUrl.ValueString(),
-	})
-	if err != nil {
-		resp.Diagnostics.AddError("Client error", fmt.Sprintf("Unable to create new API client: %s", err))
-	}
+    if p.defaultTags != nil {
+        dfTags = p.defaultTags
+    } else {
+        dfTags = utils.DefaultTagsHandler(flex.ExpandFrameworkMapString(ctx, data.DefaultTags, &resp.Diagnostics), &resp.Diagnostics)
```

Comment:
This is doing a double conversion. From `map[string]string` to `map[string]interface{}` and back to `map[string]string`

You don't need the DefaultTagsHandler, and can instead do the conversion directly. 
Also, here p.defaultTags takes precedence. IMO, the configured value should still take precendence. If configured value is not given, p.defaultTags should be used. 

```
    dfTags := p.defaultTags
    if !data.DefaultTags.IsNull() && !data.DefaultTags.IsUnknown() {
        dfTags := make(map[string]string, len(data.DefaultTags.Elements()))
        resp.Diagnostics.Append(data.DefaultTags.ElementsAs(ctx, &dfTags, false)...)
    }
```